packages/expo-updates/src/UseUpdatesUtils.ts

```diff
@@ -0,0 +1,119 @@
+import * as Updates from './Updates';
+import type { UpdateEvent } from './Updates.types';
+import type {
+  AvailableUpdateInfo,
+  CurrentlyRunningInfo,
+  Manifest,
+  UpdatesInfo,
+  UseUpdatesCallbacksType,
+} from './UseUpdates.types';
+
+// The currently running info, constructed from Updates constants
+export const currentlyRunning: CurrentlyRunningInfo = {
+  updateId: Updates.updateId,
+  channel: Updates.channel,
+  createdAt: Updates.createdAt,
+  isEmbeddedLaunch: Updates.isEmbeddedLaunch,
+  isEmergencyLaunch: Updates.isEmergencyLaunch,
+  manifest: Updates.manifest,
+  runtimeVersion: Updates.runtimeVersion,
+};
+
+/////// Internal functions ////////
+
+// Constructs the availableUpdate from the update manifest
+export const availableUpdateFromManifest = (manifest: Manifest | null | undefined) => {
```

Comment:
```suggestion
export const availableUpdateFromManifest = (manifest: Manifest) => {
```

(see second half of comment at https://github.com/expo/expo/pull/21880#discussion_r1154992576)

essentially we can better type `UpdateEvent` and other types to make it clearer when certain fields are null and others are not.

```
export type ClassicManifest = NonNullable<typeof Constants.manifest>;

export type Manifest = ClassicManifest | NonNullable<typeof Constants.manifest2>;
```

```
export type UpdateEvent =
  | {
      /**
       * Type of the event.
       */
      type: UpdateEventType.ERROR;
      /**
       * If `type` is `Updates.UpdateEventType.ERROR`, the error message, and `undefined` otherwise.
       */
      message: string;
    }
  | {
      /**
       * Type of the event.
       */
      type: UpdateEventType.UPDATE_AVAILABLE;
      /**
       * If `type` is `Updates.UpdateEventType.UPDATE_AVAILABLE`, the manifest of the newly downloaded
       * update, and `undefined` otherwise.
       */
      manifest: Manifest;
    }
  | {
      /**
       * Type of the event.
       */
      type: UpdateEventType.NO_UPDATE_AVAILABLE;
    };
```

Reply:
Fixed